app/controllers/katello/api/v2/repository_sets_controller.rb

```diff
@@ -236,18 +241,24 @@ def sort_score(pc) # sort order for enabled
               else
                 0
               end
-      Rails.logger.debug [pc.product_name, pc.enabled_content_override, "Id: #{pc.id}", "Score: #{score}"]
       score
     end
 
-    def custom_sort_results(product_content_finder)
-      if params[:sort_by] == 'enabled_by_default' && params[:sort_order] == 'desc'
-        product_content_finder.sort { |pca, pcb| sort_score(pca) <=> sort_score(pcb) }.reverse!
-      elsif params[:sort_by] == 'enabled_by_default'
-        product_content_finder.sort { |pca, pcb| sort_score(pca) <=> sort_score(pcb) }
-      else
-        product_content_finder
-      end
+    def custom_sort_results(unsorted_relation)
+      return unsorted_relation unless params[:sort_by] == 'enabled_by_default'
+      product_content_finder = ProductContentFinder.wrap_with_overrides(
+        product_contents: unsorted_relation,
+        overrides: @consumable&.content_overrides,
+        status: params[:status])
+      sorted_pcps = if params[:sort_by] == 'enabled_by_default' && params[:sort_order] == 'desc'
+                      product_content_finder.sort { |pca, pcb| sort_score(pca) <=> sort_score(pcb) }.reverse!
+                    elsif params[:sort_by] == 'enabled_by_default'
+                      product_content_finder.sort { |pca, pcb| sort_score(pca) <=> sort_score(pcb) }
+                    else
+                      product_content_finder
+                    end
+      sort_order = sorted_pcps.map(&:id)
+      unsorted_relation.reorder(Arel.sql("array_position('{#{sort_order.join(',')}}'::int[], #{Katello::ProductContent.table_name}.id)"))
```

Comment:
This only worked once I realized that I needed to call `reorder` instead of `order`, since that eliminates a default `ORDER BY` that was causing the custom sort to be ignored.

Reply:
I'm a bit  confused about how this is working. `array_position` is supposed to return a single position of an element found in an array.  But here, instead of an element, a table's ID field is being passed in as the second argument.

Reply:
`{#{sort_order.join(',')}}'::int[]` <-- So this creates a sorted array of product_content IDs, sorted in the way we want them sorted. The second argument is saying, "find where this ID is in that array, and then sort by that."

Reply:
So for example, it'll make an array of IDs like
```
[8142, 8132, 726]
```
and then the SQL will be like
```sql
SELECT blah FROM table ORDER BY array_position('{#{sort_order.join(',')}}'::int[], #{Katello::ProductContent.table_name}.id)
```

which evaluates to something like
```sql
SELECT blah FROM table ORDER BY array_position([8142, 8132, 726], katello_product_contents.id)
```

Reply:
> Instead, would it be possible to ensure that `relation` is actually a relation when it's sent to `scoped_search`? Then, the ordering can be applied as usual. It's a little odd that `relation` isn't actually a relation by the time it is passed into `scoped_search` anyway (right?).

`relation` is always a relation. I made sure of this, since that's what `scoped_search` needs. Here's the journey it goes through:

1. `index_relation` is a relation at the end of the method. It may or may not continue onto `index_relation_with_consumable_overrides`. If it does,
2. During `index_relation_with_consumable_overrides`, it makes `ProductContentFinder`s to adorn the items with the necessary override data. It then converts it back to a relation before exiting the method.
3. It's then passed into `scoped_search` as a relation, as required.

The reason all this is necessary is because

1. Content overrides are associated with hosts or activation keys, but this controller is for product contents, which is _very_ far away from those in terms of database tables.
2. Also, content overrides are a regular Ruby class, not a model. This is because they're just a snippet of Candlepin data.
3. Because of the above, there isn't a viable join we can do to "sort normally." But we already have the ProductContentFinder and ProductContentPresenter which addresses these two issues, so it works out.

Reply:
I'm convinced! Thanks for the details.